source/common/config/grpc_mux_impl.h

```diff
@@ -10,10 +10,24 @@
 #include "envoy/upstream/cluster_manager.h"
 
 #include "common/common/logger.h"
+#include "common/common/token_bucket.h"
 
 namespace Envoy {
 namespace Config {
 
+namespace {
+
+// Wrapper class for ENVOY_LOG. This is passed in GrpcMuxImpl so that critical warnings can be
+// tested.
+struct GrpcMuxLoggerImpl : public GrpcMuxLogger, Logger::Loggable<Logger::Id::upstream> {
+  void warn(const std::string& msg) const override { ENVOY_LOG(warn, "{}", msg); }
+};
+
+// Default GrpcMuxLoggerImpl instance.
+const GrpcMuxLoggerImpl GrpcMuxLogging{};
```

Comment:
This is basically a static var, which we don't allow for non-POD. It should  probably just be a constructor arg (with sensible default constructor arg).